### DISCUSSION TITLE
Variant status should be disabled if the product was disabled.

### DISCUSSION THREAD

**Discussion by alexanderloewe** [craftcms/commerce#2456] [Ideas]
**Description**
I have many products which have variants. Now I have disabled the products (the product and not the variants), yet all the variants of the disabled product are still shown in the product feed (category related).

It would be good to customize it as follows: Products which are disabled also affect all variants which are in the product group.

**Additional info**
- Craft version: 3.6.6
- Commerce version: 3.2.14.1
- PHP version: 7.4

**Comment by lukeholder** [comment]
@alexanderloewe 

Will discuss adding this as the default for variant queries, which would be a breaking change to how some people may be using variant queries at the moment.

In the meantime you can do this in your variant query:

`{% set variants = category.yourVariantRelationCustomField.hasProduct(craft.products).all() %}`

or

`{% set variants = craft.variants.hasProduct(craft.products).all() %}`

This will get all variants that have a product that is enabled since `craft.products` returns a product query for enabled products only by default. (You could add more params to the product query if you wanted)

Hope that helps.

**Comment by alexanderloewe** [comment]
I will update the code.

Nevertheless, it would make much more sense to adjust the behavior as suggested. 
I think a simple and slick way like in Craft CMS will make it – by the way the CMS is very well done from an UX perspective.
